recording/docker-compose/dockerfile-talk-recording

```diff
@@ -0,0 +1,39 @@
+FROM phusion/baseimage:focal-1.2.0
+
+# https://github.com/nextcloud/spreed/blob/master/recording/start-container.sh
+
+RUN apt-get -y update
+RUN apt-get -y upgrade
+
+# Common dependencies
+RUN apt-get -y install jq wget
+
+# spreed-recording dependencies
+RUN apt-get -y install ffmpeg pulseaudio python3-pip xvfb
+
+# firefox
+RUN apt-get -y install firefox firefox-geckodriver
+
+# chromium
+RUN add-apt-repository -y ppa:phd/chromium-browser
+RUN printf "Package: *\nPin: release o=LP-PPA-phd-chromium-browser\nPin-Priority: 1001" > /etc/apt/preferences.d/phd-chromium-browser
+RUN apt-get -y install chromium-browser chromium-chromedriver
+
+# spreed-recording config
+RUN useradd --create-home recording
+
+RUN curl -s https://api.github.com/repos/nextcloud/spreed/releases/latest | jq --raw-output '.tarball_url' | wget -i - -O /tmp/spreed-latest.tar.gz
```

Comment:
Rather than downloading the latest release, would it be possible to just traverse up in the directory tree and install the recording server from the parent directory? This way it would be possible to use a specific version by checking out the appropriate branch/tag/commit, rather than always using the latest release.

Reply:
I understand that the goal here would be being able to use a specific branch/tag/commit?

Would a [sparse checkout](https://stackoverflow.com/a/13738951) + branch/tag/commit be ok?

Reply:
Reading this after eating seems like I misunderstood the suggestion.

Currently, I have a directory that is in a completely different location than spreed/Talk. I can't _traverse_ the directory tree in my scenario.

Would it be expected/required to clone the full repo in order to start the docker container? Traversing the dir tree requires it.

Reply:
Given that the Docker files are in the same repository as the code I assumed that anyone building the image would be doing so from the full repo (even if it was just a shallow clone with a single commit using `--depth 1`), and that building the image would build the recording server at the current HEAD of the repository :shrug:

Maybe an option to accomodate both ways could be to provide a parameter/environment variable/whatever to the Dockerfile and fetch the repository in the commit/branch/hash specified in that variable. If the Dockerfile is not in the tree you need to provide an explicit value, if the Dockerfile is in the tree it can default to the current HEAD (or just traverse to the parent...). Although I am not sure if that would go against any best practices of Dockerfiles :-)

Reply:
That seems like an ugly hack to accommodate my use-case 😆

Would prefer to expect/require cloning the full repo, and always fetch the required files from the current `HEAD` (via directory traversal). Seems like an appropriate approach?

Reply:
> That seems like an ugly hack to accommodate my use-case laughing

It is :-P

> Would prefer to expect/require cloning the full repo, and always fetch the required files from the current `HEAD` (via directory traversal). Seems like an appropriate approach?

Totally. That was my initial idea before drifting into the ugly hack :-)